lib/logstash/pluginmanager/install.rb

```diff
@@ -1,82 +1,128 @@
+  # one plugin name can be also specified.
+  #
+  # TODO: find right syntax to allow specifying list of plugins with optional version specification for each
 
-  option "--version", "VERSION", "version of the plugin to install", :default => ">= 0"
+  def execute
+    if development?
+      unless plugin_list.empty?
+        $stderr.puts("Cannot specify plugin(s) with --development, it will add the development dependencies of the currently installed plugins")
+        exit(1)
+      end
+    else
+      if plugin_list.empty? && !force?
+        $stderr.puts("No plugin specified")
+        exit(1)
+      end
+      if version && plugin_list.size > 1
+        # temporary until we fullfil TODO ^^
+        $stderr.puts("Only 1 plugin name can be specified with --version")
+        exit(1)
+      end
+    end
 
-  option "--proxy", "PROXY", "Use HTTP proxy for remote operations"
+    unless File.writable?(LogStash::Environment::GEMFILE_PATH)
+      $stderr.puts("File #{LogStash::Environment::GEMFILE_PATH} does not exist or is not writable, aborting")
+      exit(1)
+    end
 
-  def execute
-    LogStash::Environment.load_logstash_gemspec!
+    gemfile = LogStash::Gemfile.new(File.new(LogStash::Environment::GEMFILE_PATH, "r+")).load
+    # keep a copy of the gemset to revert on error
+    original_gemset = gemfile.gemset.copy
 
-    ::Gem.configuration.verbose = false
-    ::Gem.configuration[:http_proxy] = proxy 
+    # force Rubygems sources to our Gemfile sources
+    Gem.sources = gemfile.gemset.sources
 
-    puts ("validating #{plugin} #{version}")
+    # install_list will be an array of [plugin name, version] tuples, version can be nil
+    install_list = []
 
-    unless gem_path = (plugin =~ /\.gem$/ && File.file?(plugin)) ? plugin : LogStash::PluginManager::Util.download_gem(plugin, version)
-      $stderr.puts ("Plugin does not exist '#{plugin}'. Aborting")
-      return 99
-    end
+    if development?
+      specs = LogStash::PluginManager.all_installed_plugins_gem_specs(gemfile)
+      install_list = specs.inject([]) do |result, spec|
+        result = result + spec.dependencies.select{|dep| dep.type == :development}.map{|dep| [dep.name] + dep.requirement.as_list + [{:group => :development}]}
+      end
+    else
+      # at this point we know that plugin_list is not empty and if the --version is specified there is only one plugin in plugin_list
+
+      install_list = version ? [plugin_list << version] : plugin_list.map{|plugin| [plugin, nil]}
 
-    unless gem_meta = LogStash::PluginManager::Util.logstash_plugin?(gem_path)
-      $stderr.puts ("Invalid logstash plugin gem '#{plugin}'. Aborting...")
-      return 99
+      install_list.each do |tuple|
+        puts("Validating #{tuple.compact.join("-")}")
+        unless LogStash::PluginManager.is_logstash_plugin?(*tuple)
+          $stderr.puts("Installation aborted")
+          exit(1)
+        end
+      end unless force?
```

Comment:
I am ... not a fan of putting an inline condition at the end of a long block. Recommend refactoring this to say:

```ruby
verify(install_list) if verify?
```

And moving the verification/validation code to a separate method.
* Assuming we rename `--force` to `--verify`